packages/app-desktop/services/autoUpdater/AutoUpdaterService.ts

```diff
@@ -58,56 +64,63 @@ export default class AutoUpdaterService implements AutoUpdaterServiceInterface {
+			throw new Error(`The AutoUpdaterService does not support the following platform: ${platform}`);
+		}
+
+		const asset: GitHubReleaseAsset = release.assets.find(a => a.name === assetName);
+		if (!asset) {
+			throw new Error('No suitable update asset found for this platform.');
+		}
+
+		return asset.browser_download_url;
+	}
+
+	private fetchReleases = async (includePreReleases: boolean): Promise<GitHubRelease[]> => {
 		const response = await fetch(releasesLink);
 
 		if (!response.ok) {
 			const responseText = await response.text();
 			throw new Error(`Cannot get latest release info: ${responseText.substr(0, 500)}`);
 		}
 
-		return (await response.json()) as GitHubRelease[];
+		const releases: GitHubRelease[] = await response.json();
+		const sortedReleasesByVersion = releases.sort((a, b) => semver.rcompare(a.tag_name, b.tag_name));
+		const filteredReleases = sortedReleasesByVersion.filter(release => includePreReleases || !release.prerelease);
+
+		return filteredReleases;
 	};
 
-	private fetchLatestRelease = async (): Promise<void> => {
+	private checkForLatestRelease = async (): Promise<void> => {
 		try {
-			const releases = await this.fetchLatestReleases();
-
-			const sortedReleasesByVersion = releases.sort((a, b) => {
-				return semver.rcompare(a.tag_name, b.tag_name);
-			});
-			const filteredReleases = sortedReleasesByVersion.filter(release => {
-				return this.includePreReleases_ || !release.prerelease;
-			});
-			const release = filteredReleases[0];
-
-			if (release) {
-				let assetUrl = null;
-
-				if (shim.isWindows()) {
-					const asset = release.assets.find((asset: GitHubReleaseAsset) => asset.name === 'latest.yml');
-					if (asset) {
-						assetUrl = asset.browser_download_url.replace('/latest.yml', '');
-					}
-				} else if (shim.isMac()) {
-					const asset = release.assets.find((asset: GitHubReleaseAsset) => asset.name === 'latest-mac.yml');
-					if (asset) {
-						assetUrl = asset.browser_download_url.replace('/latest-mac.yml', '');
-					}
-				}
-
-				if (assetUrl) {
-					autoUpdater.setFeedURL({
-						provider: 'generic',
-						url: assetUrl,
-					});
-					await autoUpdater.checkForUpdates();
-				} else {
-					this.logger_.error('No suitable update asset found for this platform.');
-				}
+			const release: GitHubRelease = await this.fetchLatestRelease(this.includePreReleases_);
+
+			try {
+				let assetUrl = this.getDownloadUrlForPlatform(release, shim.platformName());
+				// electron's autoUpdater appends automatically the platform's yml file to the link so we should remove it
+				assetUrl = assetUrl.substring(0, assetUrl.lastIndexOf('/'));
+				autoUpdater.setFeedURL({ provider: 'generic', url: assetUrl });
+				await autoUpdater.checkForUpdates();
+			} catch (error) {
+				this.logger_.error(`Update download url failed: ${error.message}`);
 			}
+
 		} catch (error) {
-			this.logger_.error(error);
+			this.logger_.error(`Fetching releases failed:  ${error.message}`);
```

Comment:
This pull request includes a lot more than just test - so would it be possible to describe what the PR is about?

Reply:
This PR only includes the tests and the refactoring of the class so I could test ferching the right version, nothing more.

On that particular line, I chose to be mor explicit when throwing an error